compiler/rustc_query_system/src/query/job.rs

```diff
@@ -540,7 +542,11 @@ pub fn deadlock(query_map: QueryMap, registry: &rayon_core::Registry) {
     // X to Y due to Rayon waiting and a true dependency from Y to X. The algorithm here
     // only considers the true dependency and won't detect a cycle.
     if !found_cycle {
-        panic!("deadlock detected");
+        if query_map.len() == 0 {
+            panic!("deadlock detected without any query!")
+        } else {
+            panic!("deadlock detected! current query map:\n{:?}", query_map);
```

Comment:
One nit, take it or leave it:

```suggestion
            panic!("deadlock detected! current query map:\n{:#?}", query_map);
```

Using the `#` alternative selector will make it so that the debug output is indented.

Reply:
Thanks! Done